crates/ruff_linter/src/rules/flake8_comprehensions/snapshots/ruff_linter__rules__flake8_comprehensions__tests__C409_C409.py.snap

```diff
@@ -239,3 +244,146 @@ C409.py:26:6: C409 [*] Unnecessary `list` literal passed to `tuple()` (rewrite a
 25 25 | t7 = tuple((1,))
 26    |-t8 = tuple([1,])
    26 |+t8 = (1,)
+27 27 | 
+28 28 | tuple([x for x in range(5)])
+29 29 | tuple({x for x in range(10)})
+
+C409.py:28:1: C409 [*] Unnecessary `list comprehension` literal passed to `tuple()` (remove the outer call to `tuple()`)
+   |
+26 | t8 = tuple([1,])
+27 | 
+28 | tuple([x for x in range(5)])
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ C409
+29 | tuple({x for x in range(10)})
+30 | tuple(x for x in range(5))
+   |
+   = help: Remove outer `tuple` call
+
+ℹ Unsafe fix
+25 25 | t7 = tuple((1,))
+26 26 | t8 = tuple([1,])
+27 27 | 
+28    |-tuple([x for x in range(5)])
+   28 |+tuple(x for x in range(5))
+29 29 | tuple({x for x in range(10)})
+30 30 | tuple(x for x in range(5))
+31 31 | tuple([
+
+C409.py:29:1: C409 [*] Unnecessary `set comprehension` literal passed to `tuple()` (remove the outer call to `tuple()`)
+   |
+28 | tuple([x for x in range(5)])
+29 | tuple({x for x in range(10)})
```

Comment:
This is not a safe transform, a set could be used here for deduplicating elements in the iterable before the tuple is created. I'm pretty sure this is only safe / sensible to flag for list comprehensions.